edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/authentication/AuthAgentProtocolHead.cs

```diff
@@ -2,27 +2,34 @@
 namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
 {
     using System;
+    using System.Collections.Generic;
+    using System.Text;
     using System.Threading;
     using System.Threading.Tasks;
     using Microsoft.AspNetCore;
     using Microsoft.AspNetCore.Hosting;
     using Microsoft.Azure.Devices.Edge.Hub.Core;
     using Microsoft.Azure.Devices.Edge.Hub.Core.Identity;
     using Microsoft.Azure.Devices.Edge.Hub.Mqtt;
+    using Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter.handlers;
     using Microsoft.Azure.Devices.Edge.Util;
     using Microsoft.Extensions.DependencyInjection;
     using Microsoft.Extensions.DependencyInjection.Extensions;
     using Microsoft.Extensions.Logging;
 
-    public class AuthAgentProtocolHead : IProtocolHead
+    public class AuthAgentProtocolHead : AbstractNotificationHandler<bool>, IProtocolHead
     {
+        const string Topic = "$internal/edgehubcore";
+        static readonly string Payload = "\"AuthAgentProtocolHead started.\"";
+
         readonly IAuthenticator authenticator;
         readonly IUsernameParser usernameParser;
         readonly IClientCredentialsFactory clientCredentialsFactory;
         readonly ISystemComponentIdProvider systemComponentIdProvider;
         readonly AuthAgentProtocolHeadConfig config;
         readonly object guard = new object();
 
+        bool _started;
```

Comment:
I'm not quite understand your point. AFAIK, `StopAsync` will be called only when we bring down EHC. But if EHC is down, connection to MQTT broker is down. Even in future we allow restart, since `_started` is `false`, no message will be send until `StartAsync` was called again. Only gap is since `IMqttBrokerConnector` has no API to tell disconnection, if it connects and disconnect and after `StartAsync`, it will still try to send message through `IMqttBrokerConnector`.

Reply:
Modified to always send out notification while connected if `AuthAgentProtocolHead` is started